### DISCUSSION TITLE
Heartbeat support from servers

### DISCUSSION THREAD

**Discussion by 3v1n0** [upptime/upptime#647] [Ideas]
It would be nice to create scripts that servers could use to notify when they are alive and use such info to generate reports and notify (creating issues) if such commits are missing when expected.

Or simply using the GitHub API to make the server(s) to trigger a job that does update the heartbeat state (not sure if environment data can be passed via the api too)

**Comment by milushov** [comment]
Missing feature for sure!

**Comment by AnandChowdhary** [comment]
Good idea! I'm not sure how it will be possible without a backend listening, though, to these notifications. Currently, Upptime works within a GitHub repository so no separate server is required, but this means that we can't expose a URL where a server can inform us of healthchecks. Do you have any ideas to help us move in this direction?
